tests/integration/s2n_tls13_handshake_tests.py

```diff
@@ -48,30 +47,41 @@ def verify_hrr_random_data(server, client):
             break
 
     return result
-    
-def key_update_recv(server, client):
+
+def key_update_test(server, client):
     '''
-    This test checks that a key update can be processed by s2n. It runs three times to prove that s2n can
-    process several key updates in a row.
+    This test proves that both server and client traffic keys can be successfully updated 
+    in a conversation between Openssl and s2n. It runs three times to confirm that multiple
+    keyupdates can be performed in a row. 
     '''
     result = Result()
     result.status = Status.PASSED
     for i in range(3):
-        msg = "Message:" + str(uuid.uuid4())
-        client.stdin.write(("k\n\n").encode("utf-8"))
+        openssl_msg = "Message:" + str(uuid.uuid4())
+        s2n_msg = "Message:" + str(uuid.uuid4())
+        # 'K' triggers Openssl to send a KeyUpdate message
```

Comment:
Three things should be tested here: (1) openssl client sends 'k': key update, (2) openssl client sends 'K': key update & request one back, (3) s2n server sends a key update message (w/out requesting one back).

Reply:
The integ test framework can definitely support both (1) and (2), but as for (3), that would require being able to alter the connection (essentially change the encrypted_bytes_out field), which we can't do. The best way to do this would be to send a bunch of data until we reach the encrypted bytes limit, but that is not conceivable for an integ test.

Reply:
"The best way to do this would be to send a bunch of data until we reach the encrypted bytes limit," ...Or add a function to let the caller force a key update, which we previously decided not to do as there is not, as yet, any real customer demand for it.
I'm good with supporting (1) and (2).